pallets/phala/src/stakepool.rs

```diff
@@ -45,8 +99,80 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		/// Creates a new stake pool
 		#[pallet::weight(0)]
-		pub fn create(origin: OriginFor<T>, id: u64) -> DispatchResult {
-			panic!("unimplemented")
+		pub fn create(origin: OriginFor<T>) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+
+			let pid = PoolCount::<T>::get();
+			PoolCount::<T>::mutate(|id| *id = pid + 1);
+			MiningPools::<T>::insert(
+				pid + 1,
+				PoolInfo {
+					pid: pid + 1,
+					owner: owner.clone(),
+					state: PoolState::default(),
+					payout_commission: Zero::zero(),
+					owner_reward: Zero::zero(),
+					pool_acc: Zero::zero(),
+					last_reward_block: T::BlockNumber::zero(),
+					total_staked: Zero::zero(),
+					extra_staked: Zero::zero(),
+					workers: vec![],
+				},
+			);
+			Self::deposit_event(Event::<T>::PoolCreated(owner, pid));
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn add_worker(
+			origin: OriginFor<T>,
+			pid: u64,
+			pubkey: WorkerPublicKey,
+		) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+
+			// make sure the worker has registered
+			ensure!(
+				registry::Worker::<T>::contains_key(&pubkey),
+				Error::<T>::WorkerNotRegistered
+			);
+			// check the wheather the owner was bounded as operator
+			let worker_info = registry::Worker::<T>::get(&pubkey).unwrap();
+			ensure!(
+				worker_info.operator == Some(owner.clone()),
+				Error::<T>::UnauthorizedOperator
+			);
```

Comment:
```suggestion
			let worker_info = registry::Worker::<T>::get(&pubkey).ok_or(Error::<T>::WorkerNotRegistered)?;
			// check the work has finished the benchmark
			ensure!(
				worker_info.intial_score.is_some(),
				Error::<T>::BenchmarkNotReady
			)
			// check the wheather the owner was bounded as operator
			ensure!(
				worker_info.operator == Some(owner.clone()),
				Error::<T>::UnauthorizedOperator
			);
```

Also need to check if the worker has the benchmark score to avoid edge cases